### PR TITLE
ARIA alerts: clarify meaning of static vs dynamic

### LINKED ISSUE #13422: Content bug: conflicting statements for intended use of ARIA alert role
## What page(s) did you find the problem on?

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role

## Specific page section or heading?

Description section: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role#description

## What is the problem?

The third paragraph of the Description section starts with: "The alert role should only be used for static text content." A few paragraphs after, the sixth paragraph starts with: "The most important thing to know about the alert role is that it's for dynamic content." These statements conflict each other and cause confusion.

## What did you expect to see?

Based on what's written in [the WAI-ARIA 1.2 docs for the alert role](https://www.w3.org/TR/wai-aria-1.2/#alert), and the same document's [definition of Live Region](https://www.w3.org/TR/wai-aria-1.2/#dfn-live-region) that says they "are typically updated", it appears the dynamic content is allowed.

## Did you test this? If so, how?

N/A

### PR DESCRIPTION
**Pull request by rightsaidjames**
Fixes #13422 - clarifies meaning of 'static text content' vs 'dynamic content' - the former refers to the type of content (non-interactive), while the latter refers to how/when the content is displayed.

#### Summary
On the a11y Slack, @ashleemboyer highlight the apparent contradiction between these two sentences:
- "The alert role should only be used for static text content."
- "The most important thing to know about the alert role is that it's for dynamic content.

She raised #13422 to document this issue.

#### Motivation
To help readers better understand:

- What type of elements the alert role should be used on.
- In what contexts the alert role should be used.

#### Related issues
#13422 

#### Supporting details

Explanation from @backwardok on the a11y Slack:

> Here’s my understanding of the sentences (although I think it’s confusingly written):
> - “The alert role should only be used for static text content.“: Don’t put interactive elements in here, only text
> - “The most important thing to know about the alert role is that it’s for dynamic content.“: Don’t add content to an alert role on page load, since it’s going to announce when it’s shown. Content should be added to an alert role container when relevant

#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [x] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/accessibility/aria/roles/alert_role/index.md [hunk 1]
```diff
@@ -24,15 +24,15 @@ The alert role should only be used for information that requires the user's imme
 - The user's login session is about to expire
 - The connection to the server was lost so local changes will not be saved
 
-The `alert` role should only be used for static text content. The element with the `alert` role does not have to be able to receive focus, as screen readers (speech or braille) will automatically announce the updated content regardless of where keyboard focus when the role is added.
+The `alert` role should only be used for static text content, not interactive elements such as links or buttons. The element with the `alert` role does not have to be able to receive focus, as screen readers (speech or braille) will automatically announce the updated content regardless of where keyboard focus when the role is added.
```
Comment (backwardok) [inline files/en-us/web/accessibility/aria/roles/alert_role/index.md:27]: I think the word "static" is part of the confusion, and the addition of the "not interactive..." aspect you added I think helps better clarify what shouldn't go in the element

```suggestion
The `alert` role should only be used for text content, not interactive elements such as links or buttons. The element with the `alert` role does not have to be able to receive focus, as screen readers (speech or braille) will automatically announce the updated content regardless of where keyboard focus when the role is added.
```

File: files/en-us/web/accessibility/aria/roles/alert_role/index.md [hunk 2]
```diff
@@ -24,15 +24,15 @@ The alert role should only be used for information that requires the user's imme
...
 
-The most important thing to know about the `alert` role is that it's for dynamic content. It is perfect for situations such as when a user fills out a form and JavaScript is used to add an error message - the alert would immediately read out the message. It should not be used on HTML that the user hasn't interacted with. For example, if a page loads with multiple visible alerts scattered throughout, no alert would be read as they were not dynamically triggered.
+The most important thing to know about the `alert` role is that it's for dynamic content that is displayed after the page has loaded. It is perfect for situations such as when a user fills out a form and JavaScript is used to add an error message - the alert would immediately read out the message. It should not be used on HTML that the user hasn't interacted with. For example, if a page loads with multiple visible alerts scattered throughout, no alert would be read as they were not dynamically triggered.
```
Comment (backwardok) [inline files/en-us/web/accessibility/aria/roles/alert_role/index.md:35]: The first sentence might still be a little confusing on what "dynamic content" is. Suggested an alternative phrasing.

I also think the last sentence example is a little confusing, so I made a suggestion on the distinction between the concept of an alert and the usage of an alert role.

```suggestion
The most important thing to know about the `alert` role is that it's for content that is dynamically displayed, not for content that appears on page load. It is perfect for situations such as when a user fills out a form and JavaScript is used to add an error message - the alert would immediately read out the message. It should not be used on HTML that the user hasn't interacted with. For example, if a page loads with multiple visible alerts scattered throughout, the alert role should not be used, as the messages were not dynamically triggered.
```

Comment (rightsaidjames) [reply files/en-us/web/accessibility/aria/roles/alert_role/index.md]: Thanks for your suggestions, I've approved them 🙂

**Review by ericwbailey — APPROVED** [review]
Well, this is lovely to see! Congratulations on your first MDN contribution, @rightsaidjames rightsaidjames 🎉

**Pull request merged**